lib/FrontendTool/ReferenceDependencies.cpp

```diff
@@ -85,18 +86,24 @@ static bool declIsPrivate(const Decl *member) {
 }
 
 static bool extendedTypeIsPrivate(TypeLoc inheritedType) {
-  if (!inheritedType.getType())
+  auto type = inheritedType.getType();
+  if (!type)
     return true;
 
-  if (!inheritedType.getType()->isExistentialType()) {
+  if (!type->isExistentialType()) {
     // Be conservative. We don't know how to deal with other extended types.
     return false;
   }
 
-  SmallVector<ProtocolDecl *, 2> protocols;
-  inheritedType.getType()->getExistentialTypeProtocols(protocols);
+  auto layout = type->getExistentialLayout();
+  assert(!layout.superclass && "Should not have a subclass existential "
+         "in the inheritance clause of an extension");
```

Comment:
Do you have a test case in mind I can try? I'll address this in an upcoming patch along with the other stuff above about spare bits.